pkg/controllers/generatorstate/generatorstate_controller.go

```diff
@@ -63,38 +63,89 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		}
 		return ctrl.Result{}, err
 	}
+	defer func() {
+		err := r.Status().Update(ctx, generatorState)
+		if err != nil {
+			r.Log.Error(err, "could not update generator state status")
+		}
```

Comment:
Hm, shouldn't this Join with the external error? It could easily be missed otherwise. Or are you okay with not reflecting this error on the object state?